node_modules/col-core/lib/db.js

```diff
@@ -78,19 +78,323 @@ var init = module.exports.init = function(callback) {
+   * @property  {Boolean}       share_points            Whether the user wants to share their point total with the other users in the course
+   */
+  var User = module.exports.User = sequelize.define('user', {
+    'canvas_user_id': {
+      'type': Sequelize.INTEGER,
+      'allowNull': false
+    },
+    'canvas_course_role': {
+      'type': Sequelize.STRING,
+      'allowNull': false
+    },
+    'canvas_full_name': {
+      'type': Sequelize.STRING,
+      'allowNull': false
+    },
+    'canvas_image': {
+      'type': Sequelize.STRING,
+      'allowNull': true
+    },
+    'points': {
+      'type': Sequelize.INTEGER,
+      'allowNull': false,
+      'defaultValue': 0
+    },
+    'share_points': {
+      'type': Sequelize.BOOLEAN,
+      'allowNull': true,
+      'defaultValue': null
+    }
+  }, {
+    'underscored': true
+  });
+
+  // Each user belongs to a specific course
+  User.belongsTo(Course, {'onDelete': 'CASCADE'});
+
+  // TODO: Define these and abstract them out in constants
+  var activityTypes = ['like', 'dislike', 'comment', 'submit_assignment'];
+  var objectTypes = ['canvas_submission', 'canvas_discussion', 'asset', 'comment'];
+
+  /**
+   * The `activity` table will keep track of actions that users make in the system. This can
+   * include things such as:
+   *  - liking or disliking assets
+   *  - commenting on assets
+   *  - submitting an assignment
+   *  - posting to a Canvas discussion
+   *  - creating a board
+   *  - ...
+   *
+   * As an activity can be related to different things, it's hard to enforce referential integrity without
+   * adding lots of unused foreign keys to the table. That's why a more general approach has been taken:
+   *  - `object_id` will be an identifier (e.g., canvas assignment id)
+   *  - `object_type` will hold information about the type the id points to (e.g., a canvas assignment)
+   *
+   * @property  {String}        type                The type of the activity
+   * @property  {Number}        object_id           The id of the object that was involved in the activity. This can be a Canvas assignment id, a comment id, an asset id, etc
+   * @property  {String}        object_type         The type of object that was involved in the activity
+   */
+  var Activity = module.exports.Activity = sequelize.define('activity', {
+    'type': Sequelize.ENUM(activityTypes),
+    'object_id': Sequelize.INTEGER,
+    'object_type': Sequelize.ENUM(objectTypes)
+  }, {
+    'underscored': true
+  });
+
+  // Each activity happens within a course
+  Activity.belongsTo(Course, {'onDelete': 'CASCADE'});
+
+  // The user who triggered the activity
+  Activity.belongsTo(User, {'as': 'actor', 'onDelete': 'CASCADE'});
+
+  // The target is the user who was indirectly involved in the activity. For example,
+  // when John likes a comment from Bella, John will be the actor and Bella will be the target.
+  // When users drop out of a course they will be removed from the system. That doesn't
```

Comment:
`When users drop out of a course, they will be removed from Collabosphere. However, the users that received points through likes and comments from these users should not lose points in the Engagement Index. Therefore, the target ids will be set to null when such a user is removed from Collabosphere.`